locker/memory_locker.go

```diff
@@ -0,0 +1,153 @@
+package locker
+
+import (
+	"fmt"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/featureform/fferr"
+	"github.com/google/uuid"
+)
+
+type memoryKey struct {
+	id      string
+	key     string
+	Channel *chan error
```

Comment:
nit: identifier smell. rename to `Done`? otherwise reads like it's meant to hold an error

Reply:
nit: channel lowercase

Reply:
channels are already pass by reference: https://stackoverflow.com/questions/44351159/using-pointer-to-channel

Reply:
semantics nit: everything in go is pass-by-value. in your case Ahmad you could remove the explicit pointer in the struct, you don't really need it. When you pass the channel a copy of its pointer is used anyway, so you have access to the original ch.